packages/cli/src/command-build.js

```diff
@@ -452,6 +454,16 @@ function directoryIsActuallyPackageName(directory) {
   return !directory.startsWith('.') && !directory.startsWith('/')
 }
 
+/**
+ * @param {string} cwd
+ */
+async function getGitBranchName(cwd) {
+  const branchName = await shWithOutput(`git branch --show-current`, {
```

Comment:
In case of detached head (e.g. after `git checkout HEAD~`), the command `git branch --show-current` will return an empty string. I would suggest that in this case, we use the commit SHA.